### PR TITLE
Porting omp parallel for loops to parallel utilities

### PR DESCRIPTION
**Pull request by ashishdarekar**
**Description**
Porting OpenMP loops to Parallel Utilities

**Changelog**
-Replaced OMP loops to ParallelUtilities & VariableUtilities

### CODE REVIEW

File: kratos/processes/assign_scalar_input_to_entities_process.cpp [hunk 1]
```diff
@@ -640,12 +636,12 @@ void AssignScalarInputToEntitiesProcess<TEntity, THistorical>::InternalAssignVal
...
-        #pragma omp parallel for
-        for(int i = 0; i<number_of_entities; i++) {
-            auto it_entity = it_begin + i;
+        IndexPartition<std::size_t>(number_of_entities).for_each([&](std::size_t index){
+            auto it_entity = it_begin + index;
             SetValue(*it_entity, rVariable, Value);
-        }
+        });
+        //Or using Variable Utitlities
+        //VariableUtils().SetNonHistoricalVariable(rVariable, Value, r_entities_array);
```
Comment (ashishdarekar) [inline kratos/processes/assign_scalar_input_to_entities_process.cpp]: @philbucher can we do like this using VariableUtils

File: kratos/processes/assign_scalar_variable_to_entities_process.h [hunk 2]
```diff
@@ -199,13 +201,13 @@ class KRATOS_API(KRATOS_CORE) AssignScalarVariableToEntitiesProcess
...
-                auto it_entity = it_begin + i;
-
+            //Using Parallel Utilities
+            IndexPartition<std::size_t>(number_of_entities).for_each([&](std::size_t index){
+                auto it_entity = it_begin + index;
                 it_entity->SetValue(rVar, Value);
-            }
+            });
+            //Or using Variable Utitlities
+            //VariableUtils().SetNonHistoricalVariable(rVar, Value, r_entities_array);
```
Comment (ashishdarekar) [inline kratos/processes/assign_scalar_variable_to_entities_process.h]: @philbucher same ??

File: kratos/processes/assign_scalar_input_to_entities_process.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/processes/assign_scalar_input_to_entities_process.cpp]: yes
Then you also don't need the `if(number_of_entities != 0) {` check any more

File: kratos/processes/assign_scalar_variable_to_entities_process.h (hunk 2 above)
Comment (philbucher) [reply kratos/processes/assign_scalar_variable_to_entities_process.h]: same comment as above

File: kratos/processes/assign_scalar_input_to_entities_process.cpp [hunk 3]
```diff
@@ -101,18 +102,14 @@ void AssignScalarInputToEntitiesProcess<TEntity, THistorical>::ExecuteInitialize
         ResetValues();
 
         if(number_of_entities != 0) {
```
Comment (philbucher) [inline kratos/processes/assign_scalar_input_to_entities_process.cpp]: this check can now be removed

**Review by philbucher — COMMENTED** [review]
minor comment

File: kratos/processes/assign_scalar_variable_to_entities_process.h [hunk 4]
```diff
@@ -23,6 +23,8 @@
 #include "includes/model_part.h"
 #include "includes/kratos_parameters.h"
 #include "processes/process.h"
+#include "utilities/parallel_utilities.h"
```
Comment (philbucher) [inline kratos/processes/assign_scalar_variable_to_entities_process.h]:
```suggestion
```
unused

**Review by philbucher — APPROVED** [review]

**Pull request merged**